requirements.txt

```diff
@@ -1,20 +1,20 @@
-numpy==1.21.0
```

Comment:
Why are we removing fixed versions in the requirements file?

Reply:
we shouldn't be too prescriptive on versions. also matching the setup.py versioning

Reply:
I would just be worried that the test environment could become quite brittle and break with new releases. Maybe something like how pytorch lightning does it, where they provide upper bounds on the versions for test environment: https://github.com/Lightning-AI/lightning/tree/master/requirements, https://github.com/Lightning-AI/lightning/blob/master/requirements/pytorch/base.txt

Reply:
that's fair. My worry though, is that the package would be broken because of setup.py but the tests wouldn't show it. I think it's useful to have tests that show when compatibility has been broken, letting us then adapt either code or setup.py

Reply:
I think the idea (according to the pytorch lightning readme on the requirements file https://github.com/Lightning-AI/lightning/tree/master/requirements), is that the dependabot manages the upper bounds of the packages in the requirements file. This would mean that you would be notified of a new release and be able to test for that when that happens (rather than relying on a test being run around that time), so you would be able to catch it more quickly. Also, you would be able to test for these in a controlled way (eg if the tests break because of compatibility issues, it may not be immediately obvious which package/packages are responsible)

Reply:
that's cool https://github.com/Lightning-AI/lightning/blob/master/.github/dependabot.yml, didn't know dependabot could do that. I'll look at it in another PR though, as I'd need to experiment with it a bit